workspaces/liferay-learn-workspace/client-extensions/liferay-learn-site-initializer/site-initializer/fragments/group/liferay-learn/fragments/article-related-recipes/index.js

```diff
@@ -0,0 +1,99 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+async function createRecipeSuggestions() {
+	const articleId = document.querySelector(
+		'.article-related-recipes'
+	).dataset.articleId;
+
+	const structuredContent = await Liferay.Util.fetch(
+		`/o/headless-delivery/v1.0/sites/${Liferay.ThemeDisplay.getSiteGroupId()}/structured-contents/by-key/${articleId}`
+	).then((response) => response.json());
+
+	if (structuredContent.keywords.length) {
+		createRecipesContainer();
+		await getRecipeKeywordsSearch(structuredContent.keywords);
+	}
+}
+
+async function getRecipeKeywordsSearch(articleKeywords) {
+	const searchParams = new URLSearchParams({
+		fields: 'friendlyUrlPath,title,datePublished,description',
+		search: articleKeywords.slice(0, articleKeywords.length).join(","),
+		sort: 'datePublished:desc',
+		pageSize: '3',
+	}).toString();
+
+	const response = await Liferay.Util.fetch(
+		`/o/headless-delivery/v1.0/structured-content-folders/34105240/structured-contents?${searchParams}`
+	).then((response) => response.json());
+
+	const responseItems = response.items;
+
+	responseItems.forEach((item) => 
+		createRecipeLinkCard(item.title, item.datePublished, item.description, item.friendlyUrlPath)
+	);
+}
+
+function createRecipesContainer() {
```

Comment:
Instead of creating each tag, adding classes and content, you can use template literals + innerHTML to build the html structure. (follow the logic you used it `createRecipeLinkCard` function)